modules/api/src/main/java/org/apache/fluo/api/data/Column.java

```diff
@@ -28,26 +28,42 @@
 public final class Column implements Comparable<Column>, Serializable {
 
   private static final long serialVersionUID = 1L;
-  private static final Bytes UNSET = Bytes.of(new byte[0]);
 
-  private Bytes family = UNSET;
-  private Bytes qualifier = UNSET;
-  private Bytes visibility = UNSET;
+  private final Bytes family;
+  private final Bytes qualifier;
+  private final Bytes visibility;
+
+  private final boolean isFamilySet;
+  private final boolean isQualifierSet;
+  private final boolean isVisibilitySet;
```

Comment:
Instead of 6 fields, could use Optional to have only 3. Could use Optional.orElse(emptyBytes) in the getter instead of checking if the corresponding boolean is set

Reply:
I considered using Optional but decided against it because it would add more Objects to GC and another level of indirection (so much pointer chasing in java in general, I suspect this why even java compiles to native its still slower than C++).  This low level data class already has a good bit of both (objs and indirection), didn't want to add more.